SalesforceCommerceCloud/pwa-kit
# Add fallback currency to `<ProductTile />` component

# Description



The `<ProductTile />` component assumes the `product` prop will always have a `currency` value defined.

We have recently seen the Einstein API response sometimes contains products without a `currency` value.

Like in this particular product:

```
{
    "currentPrice": null,
    "id": "in-store-setM",
    "imageUrl": "https://edge.disstg.commercecloud.salesforce.com/dw/image/v2/ZZCU_007/on/demandware.static/-/Sites-apparel-m-catalog/default/dw79ab5d14/images/large/90011212_001_0.jpg",
    "productName": "In store set",
    "productUrl": "https://zzcu-007.sandbox.us01.dx.commercecloud.salesforce.com/womens/clothing/outfits/in-store-setM.html?lang=en_US",
    "productId": "in-store-setM",
    "image": {
        "disBaseLink": "https://edge.disstg.commercecloud.salesforce.com/dw/image/v2/ZZCU_007/on/demandware.static/-/Sites-apparel-m-catalog/default/dw79ab5d14/images/large/90011212_001_0.jpg",
        "alt": "In store set"
    }
}
```

In that case, the value of the `currency` parameter used by `intl.formatNumber` method is `undefined`, and thus a react intl error is presented to shoppers:

```
Error: [@formatjs/intl Error FORMAT_ERROR] Error formatting number. 

Currency code is required with currency style.
TypeError: Currency code is required with currency style.
    at new NumberFormat (<anonymous>)
    at https://pwa-kit.mobify-storefront.com/mobify/bundle/384/vendor.js:2:385338
    at i (https://pwa-kit.mobify-storefront.com/mobify/bundle/384/vendor.js:2:355864)
    at g (https://pwa-kit.mobify-storefront.com/mobify/bundle/384/vendor.js:2:844513)
    at y (https://pwa-kit.mobify-storefront.com/mobify/bundle/384/vendor.js:2:844584)
    at x (https://pwa-kit.mobify-storefront.com/mobify/bundle/384/main.js:1:129837)
    at aa (https://pwa-kit.mobify-storefront.com/mobify/bundle/384/vendor.js:2:746024)
    at Gs (https://pwa-kit.mobify-storefront.com/mobify/bundle/384/vendor.js:2:798073)
    at Ou (https://pwa-kit.mobify-storefront.com/mobify/bundle/384/vendor.js:2:785347)
    at Su (https://pwa-kit.mobify-storefront.com/mobify/bundle/384/vendor.js:2:785275)
```

The change adds the `activeCurrency` as a fallback when the product's `currency` sent to the `<ProductTile />` component is undefined.

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Added `activeCurrency` as a fallback currency in `<ProductTile />` component.

# How to Test-Drive This PR

- Run the PWA locally.
- Verify that the product currently failing now loads properly:
http://localhost:3000/en-GB/product/25686514M

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
